### PR TITLE
Add validation logic to multi-option questions

### LINKED ISSUE #416: "Dropdown" question type: implement simple validation criteria
Let the admin specify:

- Minimum number of choices that can be picked
- Maximum number of choices that can be picked

when building a question of this type.

See #395, the parent issue for this, and [this wiki page](https://github.com/seattle-uat/universal-application-tool/wiki/Question-type-requirements-notes) for details.

### LINKED ISSUE #419: "Checkbox set" question type: implement simple validation criteria
Let the admin specify:

- Minimum number of choices that can be picked
- Maximum number of choices that can be picked

when building a question of this type.

See #396, the parent issue for this, and [this wiki page](https://github.com/seattle-uat/universal-application-tool/wiki/Question-type-requirements-notes) for details.

### PR DESCRIPTION
**Pull request by carolinedanzi**
### Description
Adds validation logic to multi-option questions, which consists of:

1. Minimum number of choices that can be picked
2. Maximum number of choices that can be picked

Changes:
1. Adds multi-select validation fields in the admin view  (checkbox question only) - `MultiOptionQuestionForm` + test, `QuestionConfig`
2. Adds predicate `AutoValue` to `MultiOptionQuestionDefinition` and updates checkbox, radio, and dropdown to  specify  validation
3. Enforces validation on applicant side - `MultiSelectQuestion` + test

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #419 
Fixes #416

### CODE REVIEW

File: universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java [hunk 1]
```diff
@@ -38,6 +38,18 @@ public void preferredLocale_defaultsToEnglish() {
     assertThat(data.preferredLocale()).isEqualTo(Locale.US);
   }
 
+  @Test
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java:41]: Question (non-blocking): is this from a merge conflict or an intended change? no need to change, just curious since this seems unrelated.

File: universal-application-tool-0.0.1/test/services/applicant/question/SingleSelectQuestionTest.java [hunk 2]
```diff
@@ -58,16 +58,29 @@ public void withEmptyApplicantData() {
...
 
     assertThat(singleSelectQuestion.hasTypeSpecificErrors()).isFalse();
     assertThat(singleSelectQuestion.hasQuestionErrors()).isFalse();
-    assertThat(singleSelectQuestion.getSelectedOptionValue()).hasValue("answer");
+    assertThat(singleSelectQuestion.getSelectedOptionValue()).hasValue("option 1");
   }
 
-  // TODO(https://github.com/seattle-uat/civiform/issues/416): Add a test for validation failures.
+  @Test
+  public void withPresentApplicantData_selectedNotInvalidOption_hasErrors() {
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/test/services/applicant/question/SingleSelectQuestionTest.java:73]: Nit (blocking): NotInvalid -> Invalid

File: universal-application-tool-0.0.1/test/forms/MultiOptionQuestionFormTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import services.Path;
+import services.question.types.CheckboxQuestionDefinition;
+import services.question.types.MultiOptionQuestionDefinition;
+import services.question.types.QuestionDefinition;
+import services.question.types.QuestionDefinitionBuilder;
+
+public class MultiOptionQuestionFormTest {
+
+  @Test
+  public void getBuilder_returnsCompleteBuilder() throws Exception {
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/test/forms/MultiOptionQuestionFormTest.java:20]: Praise: nice!

**Review by hmhsu — APPROVED** [review]
LGTM!

File: universal-application-tool-0.0.1/test/services/applicant/question/SingleSelectQuestionTest.java (hunk 2 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/test/services/applicant/question/SingleSelectQuestionTest.java:73]: Oof good catch, thank you!

File: universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java (hunk 1 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java:41]: Ah I forgot about this - there was a point where I started to doubt that data was being overwritten, so I  added a test. I'll keep it in since it isn't hurting anything, but you're right it is somewhat unrelated

**Comment by carolinedanzi** [comment]
> LGTM!

Thank you for the review! I really appreciate it!

File: universal-application-tool-0.0.1/test/forms/MultiOptionQuestionFormTest.java (hunk 3 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/test/forms/MultiOptionQuestionFormTest.java:20]: Thanks!
